arrow-select/src/nullif.rs

```diff
@@ -128,6 +128,27 @@ mod tests {
         assert_eq!(&expected, res);
     }
 
+    #[test]
+    fn test_nullif_null_array() {
+        assert_eq!(
+            as_null_array(
+                &nullif(&NullArray::new(0), &BooleanArray::new_null(0)).unwrap()
+            ),
```

Comment:
```suggestion
            nullif(&NullArray::new(0), &BooleanArray::new_null(0)).unwrap().as_ref(),
```
FWIW you don't need to downcast here

Reply:
I'll update another one as well.